### DISCUSSION TITLE
Limitador de Consultas

### DISCUSSION THREAD

**Discussion by Kiny-Kiny** [Kiny-Kiny/Kiny-Painel#10] [Ideas]
Bem, eu tô pensando em colocar um limitador de Consultas em certas pesquisas como a de CPF, Nome e telefone, pelo simples fato delas poderem cair por causa do uso abusivo.

**Comment by cleitondemelod249** [comment]
Eu concordo com isso. Porque é uma ferramenta muito útil pra algumas situações e tem gente que fica abusando da API pra fins banais.

**Reply by Kiny-Kiny** [reply]
Sim, concordo, tinham várias pessoas consultando sem motivo algum.

**Comment by cleitondemelod249** [comment]
Se você tiver acesso ao lado do webservice, seria interessante um token antes de receber a requisição ou então amarrar o Mac ou IP do usuário pra ter um controle maior se possível. 
Sei lá, só ideias mesmo pra ver se consigo ajudar de alguma forma 😆

**Reply by Kiny-Kiny** [reply]
Ótima idéia, coloquei um for range por enquanto xD. Mas meio que eu não teria que tornar meio que o painel meio que um ssh para fazer isso? meio que precisaria sempre se atualizar em tempo real, como um chat online.

**Reply by cleitondemelod249** [reply]
Entendo. Essas minhas ideias só valem se tiver um controlador do lado da API.

**Reply by Kiny-Kiny** [reply]
> Entendo. Essas minhas ideias só valem se tiver um controlador do lado da API.

Sim, mas ela ela foi realmente excelente. Acho que se codar por cima possa ajudar.

**Reply by Doges** [reply]
Ideia boa porém acho que seria bom também um terceiro método, como algum id do computador que não pode ser alterado, ou se puder ser alterado que demore, como por exemplo o nome da máquina ou algo do tipo, porém se for do lado do server isso seria difícil

**Reply by cleitondemelod249** [reply]
Da pra fazer isso muito fácil do lado da aplicação. O problema é que os mais espertos poderiam ir atrás do link da API dentro do código e usar livremente. 

Por isso que o mais importante é fazer esses tipos de validações no lado do webservice.

**Reply by oporadokrl** [reply]
@Kiny-Kiny precisamos buscar os donos das APIs pra isso,como dito anteriormente só o dev da API pode fazer esse block de requisição, se fosse um bot de telegram seria mais fácil.

**Reply by Kiny-Kiny** [reply]
@oporadokrl Sim, acho melhor a gente dar uma olhada depois.

**Comment by narlles** [comment]
na minha opniao o limitador de consultas nao faria sentido algum. Pelo fato de que o projeto se trata disso

**Reply by Kiny-Kiny** [reply]
Amigo, o problema é que as APIs pagas aqui caem rápido demais, pelo único fato de não ter um limitador.

**Reply by Kiny-Kiny** [reply]
Conseguimos derrubar até uma API gratuita só pelo tanto de consulta que fizeram.

**Comment by vinipessanha** [comment]
Seria interessante exibir o telefone e outros dados, como empresa, etc.

**Reply by Kiny-Kiny** [reply]
Estamos fazendo isso já;D
